examples/ogbn_papers_100m.py

```diff
@@ -119,17 +212,68 @@ def test(loader: NeighborLoader, val_steps: Optional[int] = None):
     return total_correct / total_examples
 
 
-torch.cuda.synchronize()
+test_accs = []
+val_accs = []
+times = []
+train_times = []
+inference_times = []
+best_val = best_test = 0.
 prep_time = round(time.perf_counter() - wall_clock_start, 2)
 print("Total time before training begins (prep_time)=", prep_time, "seconds")
 print("Beginning training...")
-for epoch in range(1, 1 + args.epochs):
-    train()
-    val_acc = test(val_loader, val_steps=100)
-    print(f'Val Acc: ~{val_acc:.4f}')
+for run in range(1, num_runs + 1):
+    start = time.time()
+    print(f'\nRun {run:02d}:\n')
+
+    model.reset_parameters()
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
+
+    best_val_acc = best_test_acc = 0.0
+    for epoch in range(1, num_epochs + 1):
+        train_start = time.time()
+        loss, acc = train(epoch)
+        train_end = time.time()
+        train_times.append(train_end - train_start)
+        print(f'Epoch {epoch:02d}, Loss: {loss:.4f}, Approx. Train: {acc:.4f}'
+              f' Time: {train_end - train_start:.4f}s')
+
+        inference_start = time.time()
+        train_acc = test(test_loader)
+        val_acc = test(val_loader)
+        test_acc = test(test_loader)
```

Comment:
Evaluating the test accuracy during training, and picking up the best "test accuracy" looks incorrect to me.